@navikt/core/react/src/form/combobox/FilteredOptions/filteredOptionsContext.tsx

```diff
@@ -136,9 +137,12 @@ const FilteredOptionsProvider = ({
   const toggleIsListOpen = useCallback(
     (newState?: boolean) => {
       virtualFocus.moveFocusToTop();
+      if (newState ?? !isInternalListOpen) {
+        setHideCaret(!!maxSelected?.isLimitReached);
+      }
       setInternalListOpen((oldState) => newState ?? !oldState);
     },
-    [virtualFocus],
+    [virtualFocus, maxSelected, isInternalListOpen, setHideCaret],
```

Comment:
Since `maxSelected` is an object generated on each render, it will be "new" on each render. Consider having `maxSelected?.isLimitReached` instead of `maxSelected` in the dependency array.
```suggestion
    [
      virtualFocus,
      maxSelected?.isLimitReached,
      isInternalListOpen,
      setHideCaret,
    ],
```